controllers/storagecluster/readiness.go

```diff
@@ -16,6 +26,17 @@ var ReadinessChecker healthz.Checker = func(_ *http.Request) error {
 	return errors.New("StorageCluster is not ready yet")
 }
 
+func SetOperatorConditions(c conditions.Condition, reason string, message string, ready bool) {
+	if ready {
```

Comment:
This function does not seem to be required. The OperatorCondition should not update the operatorReady variable as that would cause the readiness probe to fail if the Upgradeable condition is false. And we do not want that

Reply:
Since we are not calling ReadinessSet() and ReadinessUnset() , how do we set the operatorReady variable for readiness probe?

Reply:
It should be set to true once the operator is up and then left as it is - we do not need to change it depending on the storagecluster creation or we end up with the same problem where the CSV will move to failed.

Reply:
So that means we only call ReadinessSet() once in main method and leave the rest of status handling by OperatorConditions?